xyz-connectors/src/main/java/com/here/xyz/connectors/AbstractConnectorHandler.java

```diff
@@ -157,18 +157,21 @@ public void handleRequest(InputStream input, OutputStream output, Context contex
    * @return the event
    */
   Event readEvent(InputStream input) throws ErrorResponseException {
+    String streamPreview = null;
     try {
       input = Payload.prepareInputStream(input);
-      String streamPreview = previewInput(input);
+      streamPreview = previewInput(input);
 
       Event receivedEvent = XyzSerializable.deserialize(input);
       logger.info("{} [{} ms] - Parsed event: {}", receivedEvent.getStreamId(), ms(), streamPreview);
       return receivedEvent;
-    } catch (ClassCastException e) {
-      logger.error("{} [{} ms] - Exception occurred while reading the event: {}", "FATAL", ms(), e);
-      throw new ErrorResponseException(streamId, XyzError.ILLEGAL_ARGUMENT, "The input is doesn't have known type.");
     } catch (Exception e) {
-      logger.error("{} [{} ms] - Exception occurred while reading the event: {}", "FATAL", ms(), e);
+      logger.error("{} [{} ms] - Exception {} occurred while reading the event: {}", "FATAL", ms(), e.getMessage(), streamPreview, e);
+
+      if (e instanceof ClassCastException) {
+        throw new ErrorResponseException(streamId, XyzError.ILLEGAL_ARGUMENT, "The input is doesn't have known type.");
+      }
```

Comment:
Not sure, but I do not think that this statement will be ever true. You catch all ClassCastExceptions and sub-classes of it in the catch block above.

Reply:
Pulling out the `String streamPreview = null;` should be enough here to be able to print the original event-string as the first catch-block (in the original version) would catch the `ClassCastException`s before the 2nd catch-block.